contracts/facets/ThirdPartyFacet.sol

```diff
@@ -109,28 +112,35 @@ contract ThirdPartyFacet {
         }
 
         // Confirm the sarcophagus is not buried
-        if (sarcophagus.resurrectionTime == 2**256 - 1) {
+        if (sarcophagus.resurrectionTime == 2 ** 256 - 1) {
             revert LibErrors.SarcophagusInactive(sarcoId);
         }
 
         // build an array of the addresses of the archaeologists currently being accused
-        address[] memory accusedArchAddresses = new address[](keyShareHashes.length);
+        address[] memory accusedArchAddresses = new address[](signatures.length);
 
         // track the combined locked bond across all archaeologists being accused in this call
         // locked bond will be equal to the amount of diggingFees allocated by the embalmer to pay the archaeologist
         uint256 totalDiggingFees = 0;
         uint256 accusalCount = 0;
-        for (uint256 i = 0; i < keyShareHashes.length; i++) {
-            // generate the double hash of the key share
-            bytes32 doubleHashedKeyShare = keccak256(abi.encode(keyShareHashes[i]));
-
-            // look up the archaeologist responsible for the key share
-            address accusedArchaeologistAddress = s.doubleHashedShardArchaeologists[doubleHashedKeyShare];
-            LibTypes.CursedArchaeologist storage accusedArchaeologist = sarcophagus.cursedArchaeologists[accusedArchaeologistAddress];
-
-            // verify the accused archaeologist is cursed on the sarcophagus
-            if (accusedArchaeologist.doubleHashedKeyShare == 0) {
-                revert LibErrors.ArchaeologistNotOnSarcophagus(msg.sender);
+        for (uint256 i = 0; i < signatures.length; i++) {
+            // Recover the archaeologist's address from the signature. Note that the recoverAddress
+            // function will always return a valid address, even if the message does not match the
+            // signature.
+            address accusedArchaeologistAddress = LibUtils.recoverAddress(
```

Comment:
This is going to recover the address corresponding to the public key the archaeologist has generated for the sarcophagus, but it's not going to be the same as the archaeologist's address.

We'll need to recover the full public key from the signature, then look up the archaeologist address given that public key in the new convenience mapping we're creating (publicKey=>archaeologistAddress)